Fix double-slash disable comments followed by another comment

> Clearly describe the bug

When upgrading to 13.7.0, we found that double-slash disable comments:

`// stylelint-disable color-no-hex`

... stopped working (and started throwing errors) in 3 files out of several hundred.  Two of them were first-line, whole-file comments, the third was a disable-next-line for a specific rule.

Converting these comments to the other style fixed the issue:

`/* stylelint-disable color-no-hex */`

> Which rule, if any, is the bug related to?

N/A

> What code is needed to reproduce the bug?

Since this only occurred in a tiny sample of our files, I don't know of a specific repro path.

``


> Which version of stylelint are you using?

13.7.0

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

`CLI via npx stylelint \"{apps,libs}/**/*.scss\"`

Here is the output with one of the rules reverted to the double-slash style:
```
╰─ npm run stylelint                                                                                                                                          3528  08:28:41   

> ccxp-client@0.0.1 stylelint /Users/pete.burkindine/git/ccxp/ccxp-client
> stylelint "{apps,libs}/**/*.scss"


apps/ccxp-web/src/scss/defines/_colors.scss
   4:9   ✖  Unexpected hex color "#fff"      color-no-hex
   5:9   ✖  Unexpected hex color "#000"      color-no-hex
   8:13  ✖  Unexpected hex color "#f3f3f3"   color-no-hex
   9:13  ✖  Unexpected hex color "#d6d7d9"   color-no-hex
  10:13  ✖  Unexpected hex color "#72777f"   color-no-hex
  11:13  ✖  Unexpected hex color "#5b616b"   color-no-hex
  12:13  ✖  Unexpected hex color "#323a45"   color-no-hex
  13:13  ✖  Unexpected hex color "#1f252d"   color-no-hex
  33:12  ✖  Unexpected hex color "#e0f2f1"   color-no-hex
  34:12  ✖  Unexpected hex color "#409e90"   color-no-hex
  35:12  ✖  Unexpected hex color "#146a5d"   color-no-hex
  36:12  ✖  Unexpected hex color "#0f4b42"   color-no-hex
  56:12  ✖  Unexpected hex color "#e8f1ff"   color-no-hex
  57:12  ✖  Unexpected hex color "#4f6b9b"   color-no-hex
  58:12  ✖  Unexpected hex color "#1e3c70"   color-no-hex
  59:12  ✖  Unexpected hex color "#12284d"   color-no-hex
  79:13  ✖  Unexpected hex color "#e7f4e3"   color-no-hex
  80:13  ✖  Unexpected hex color "#94bfa2"   color-no-hex
  81:13  ✖  Unexpected hex color "#12890e"   color-no-hex
  82:13  ✖  Unexpected hex color "#0d5e0b"   color-no-hex
 102:14  ✖  Unexpected hex color "#fff1d2"   color-no-hex
 103:14  ✖  Unexpected hex color "#fad980"   color-no-hex
 104:14  ✖  Unexpected hex color "#f8c41f"   color-no-hex
 105:14  ✖  Unexpected hex color "#f1ad15"   color-no-hex
 106:21  ✖  Unexpected hex color "#d67d00"   color-no-hex
 126:11  ✖  Unexpected hex color "#f8dddd"   color-no-hex
 127:11  ✖  Unexpected hex color "#e59393"   color-no-hex
 128:11  ✖  Unexpected hex color "#b20000"   color-no-hex
 129:11  ✖  Unexpected hex color "#8d0000"   color-no-hex
 160:33  ✖  Unexpected hex color "#cfeffc"   color-no-hex
 161:29  ✖  Unexpected hex color "#046791"   color-no-hex

npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! ccxp-client@0.0.1 stylelint: `stylelint "{apps,libs}/**/*.scss"`
npm ERR! Exit status 2
npm ERR! 
npm ERR! Failed at the ccxp-client@0.0.1 stylelint script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/pete.burkindine/.npm/_logs/2020-08-31T13_31_44_626Z-debug.log
```

Here are the contents of the mentioned log file:
```
0 info it worked if it ends with ok
1 verbose cli [ '/usr/local/bin/node', '/usr/local/bin/npm', 'run', 'stylelint' ]
2 info using npm@6.14.5
3 info using node@v12.16.3
4 verbose run-script [ 'prestylelint', 'stylelint', 'poststylelint' ]
5 info lifecycle ccxp-client@0.0.1~prestylelint: ccxp-client@0.0.1
6 info lifecycle ccxp-client@0.0.1~stylelint: ccxp-client@0.0.1
7 verbose lifecycle ccxp-client@0.0.1~stylelint: unsafe-perm in lifecycle true
8 verbose lifecycle ccxp-client@0.0.1~stylelint: PATH: /usr/local/lib/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/Users/pete.burkindine/git/ccxp/ccxp-client/node_modules/.bin:/Users/pete.burkindine/bin:/usr/local/bin:/Library/PostgreSQL/10/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/MacGPG2/bin:/Users/pete.burkindine/bin:/Library/PostgreSQL/10/bin
9 verbose lifecycle ccxp-client@0.0.1~stylelint: CWD: /Users/pete.burkindine/git/ccxp/ccxp-client
10 silly lifecycle ccxp-client@0.0.1~stylelint: Args: [ '-c', 'stylelint "{apps,libs}/**/*.scss"' ]
11 silly lifecycle ccxp-client@0.0.1~stylelint: Returned: code: 2  signal: null
12 info lifecycle ccxp-client@0.0.1~stylelint: Failed to exec stylelint script
13 verbose stack Error: ccxp-client@0.0.1 stylelint: `stylelint "{apps,libs}/**/*.scss"`
13 verbose stack Exit status 2
13 verbose stack     at EventEmitter.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/npm-lifecycle/index.js:332:16)
13 verbose stack     at EventEmitter.emit (events.js:310:20)
13 verbose stack     at ChildProcess.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:55:14)
13 verbose stack     at ChildProcess.emit (events.js:310:20)
13 verbose stack     at maybeClose (internal/child_process.js:1021:16)
13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:286:5)
14 verbose pkgid ccxp-client@0.0.1
15 verbose cwd /Users/pete.burkindine/git/ccxp/ccxp-client
16 verbose Darwin 19.6.0
17 verbose argv "/usr/local/bin/node" "/usr/local/bin/npm" "run" "stylelint"
18 verbose node v12.16.3
19 verbose npm  v6.14.5
20 error code ELIFECYCLE
21 error errno 2
22 error ccxp-client@0.0.1 stylelint: `stylelint "{apps,libs}/**/*.scss"`
22 error Exit status 2
23 error Failed at the ccxp-client@0.0.1 stylelint script.
23 error This is probably not a problem with npm. There is likely additional logging output above.
24 verbose exit [ 2, true ]

```

--- a/lib/__tests__/disableRanges.test.js
+++ b/lib/__tests__/disableRanges.test.js
@@ -781,6 +781,54 @@ it('SCSS // disable next-line comment (with multi-line description)', () => {
 		});
 });
 
+it('SCSS // disable comment (with // comment after blank line)', () => {
+	const scssSource = `a {
+    // stylelint-disable declaration-no-important
+
+    // Unrelated
+    color: pink !important;
+  }`;
+
+	return postcss()
+		.use(assignDisabledRanges)
+		.process(scssSource, { syntax: scss, from: undefined })
+		.then((result) => {
+			expect(result.stylelint.disabledRanges).toEqual({
+				all: [],
+				'declaration-no-important': [
+					{
+						start: 2,
+						strictStart: true,
+					},
+				],
+			});
+		});
+});
+
+it('SCSS /* disable comment (with // comment after blank line)', () => {
+	const scssSource = `a {
+    /* stylelint-disable declaration-no-important */
+
+    // Unrelated
+    color: pink !important;
+  }`;
+
+	return postcss()
+		.use(assignDisabledRanges)
+		.process(scssSource, { syntax: scss, from: undefined })
+		.then((result) => {
+			expect(result.stylelint.disabledRanges).toEqual({
+				all: [],
+				'declaration-no-important': [
+					{
+						start: 2,
+						strictStart: true,
+					},
+				],
+			});
+		});
+});
+
 it('Less // line-disabling comment (with description)', () => {
 	const lessSource = `a {
     color: pink !important; // stylelint-disable-line declaration-no-important -- Description

--- a/lib/assignDisabledRanges.js
+++ b/lib/assignDisabledRanges.js
@@ -69,12 +69,17 @@ module.exports = function (root, result) {
 		} else if (isInlineComment(comment)) {
 			const fullComment = comment.clone();
 			let next = comment.next();
+			let lastLine = (comment.source && comment.source.end && comment.source.end.line) || 0;
 
 			while (next && next.type === 'comment') {
 				/** @type {PostcssComment} */
 				const current = next;
 
 				if (!isInlineComment(current)) break;
+
+				const currentLine = (current.source && current.source.end && current.source.end.line) || 0;
+
+				if (lastLine + 1 !== currentLine) break;
 
 				fullComment.text += `\n${current.text}`;
 
@@ -84,6 +89,7 @@ module.exports = function (root, result) {
 
 				inlineEnd = current;
 				next = current.next();
+				lastLine = currentLine;
 			}
 			checkComment(fullComment);
 		} else {

